lib/getStaticTypes.js

```diff
@@ -0,0 +1,41 @@
+const findUp = require('find-up');
+const fastGlob = require('fast-glob').async;
+const keyBy = require('lodash/keyBy');
+const mapValues = require('lodash/mapValues');
+const omit = require('lodash/omit');
+
+const stringRegex = /^"(.*)"$/;
+const parsePropTypeName = propTypeName => {
+  return propTypeName
+    .split(' | ')
+    .filter(x => stringRegex.test(x))
+    .map(x => x.replace(stringRegex, '$1'));
+};
+
+const filterProps = props => omit(props, 'className', 'children');
+
+module.exports = async ({ cwd }) => {
+  try {
+    const tsConfigPath = await findUp('tsconfig.json', { cwd });
+
+    if (!tsConfigPath) {
+      return {};
+    }
+
+    const { parse } = require('react-docgen-typescript').withCustomConfig(
+      tsConfigPath
+    );
+    const files = await fastGlob([`${cwd}/**/*.{ts,tsx}`, '!**/node_modules']);
```

Comment:
Should we be respecting the `include` and `exclude` values of the `tsconfig.json` if they exist, and then fall back to this implementation if they haven't been defined?

Reply:
Found this was a bit too much of a rabbit hole right now, so added an optional `typeScriptFiles` option so that you can override the default globs.